test/unit/puppetlabs/services/jruby/jruby_testutils.clj

```diff
@@ -52,15 +54,14 @@
   JRubyPuppet pool size with be included with the config, otherwise no size
   will be specified."
   ([]
-   {:ruby-load-path  ruby-load-path
-    :gem-home        gem-home
-    :master-conf-dir conf-dir})
+   (jruby-core/initialize-config
+     {:jruby-puppet {:gem-home        gem-home
+                     :master-conf-dir conf-dir
+                     :master-var-dir  var-dir}
+      :os-settings  {:ruby-load-path ruby-load-path}}))
```

Comment:
So, this is confusing, but there are actually two different kinds of config objects in play here.

The one you commented on is basically the raw structure that you'd find in the config files (on `stable`, anyway; in `master` the `os-settings` map should be gone entirely).

The one before that is a different thing.  During the initialization of the jruby service there is some validation / transformation code:

https://github.com/cprice404/puppet-server/blob/feature/stable/SERVER-325-reorganize-code-for-max-requests/src/clj/puppetlabs/services/jruby/jruby_puppet_core.clj#L84-L103

whose job it is to take the raw config from the files and build up a map that complies with the `JRubyPuppetConfig` schema, which is what we pass around from that point forward:

https://github.com/cprice404/puppet-server/blob/feature/stable/SERVER-325-reorganize-code-for-max-requests/src/clj/puppetlabs/services/jruby/jruby_puppet_schemas.clj#L30-L64

It's confusing that we use both in the tests, but I think it's probably warranted because they're both necessary if you want to have tests at both levels of the system.